geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskRegionCompactorClearsObjectThatAreNoLongerNeededIntegrationTest.java

```diff
@@ -0,0 +1,455 @@
+    // destroys for events that are located in .crf files from previous range.
+    TombstoneService tombstoneService = ((InternalCache) cache).getTombstoneService();
+    int key = 0;
+    while (key < ENTRY_RANGE_1) {
+      region.destroy(key);
+      // It is necessary to force tombstone expiration, otherwise event won't be stored in .drf file
+      // and total live count won't be decreased
+      await().untilAsserted(
+          () -> assertThat(tombstoneService.forceBatchExpirationForTests(1)).isTrue());
+      region.put(key + ENTRY_RANGE_1, new byte[300]);
+      key = key + 5;
+    }
+    await().untilAsserted(() -> assertThat(getCurrentNumberOfOplogs(diskStore)).isEqualTo(7));
+
+    // Destroy all events created in previous step in order to trigger automatic compaction.
+    // This will trigger compaction for the files that were created in previous step.
+    // Compaction will delete .crf and .krf file, but will leave .drf file because it contains
+    // destroy operation for the events that are located in some older .crf files.
+    key = ENTRY_RANGE_1;
+    while (key < ENTRY_RANGE_2) {
+      region.destroy(key);
+      assertThat(tombstoneService.forceBatchExpirationForTests(1)).isTrue();
+      key = key + 5;
+    }
+
+    // wait for all Oplog's to be compacted
+    await().untilAsserted(() -> assertThat(isOplogToBeCompactedAvailable(diskStore)).isFalse());
+
+    await().untilAsserted(
+        () -> assertThat(areAllUnnecessaryObjectClearedForOnlyDrfOplog(diskStore)).isTrue());
+  }
+
+  /**
+   * Verifies that the unnecessary memory is cleared when operational log (.crf and .drf) is
+   * compacted.This is special scenario were creation of .krf file is cancelled by ongoing
+   * compaction. This usually happens when new oplog is rolled out and previous oplog is
+   * immediately marked as eligible for compaction. Compaction and .krf creation start at the
+   * similar time and compactor cancels creation of .krf if it is executed first.
+   *
+   * This test case covers the following scenario:
+   *
+   * 1. Create several Oplog files (.crf, .drf and .krf) by executing put operations.
+   * 2. Execute destroy operation for every fifth entry, and each time add new entry. When
+   * it is time for oplog to roll out, the previous oplog will be immediately marked as ready
+   * for compaction because compaction threshold is set to high value in this case. This way
+   * we force race condition where compaction cancel creation of .krf file. Compaction will
+   * delete only .crf file (.krf was not created at all), but will not delete .drf files because
+   * they contain destroy operations for events located in .crf files created in step 1. Check
+   * that unnecessary objects are cleared for the Oplog that represents orphaned .drf file
+   * (no accompanying .crf and .krf file)
+   **/
+  @Test
+  public void testCompactorRegionMapDeletedAfterCompactionForOnlyDrfOplogIsDoneRaceCondition()
```

Comment:
Instead of using `RaceCondition` in the test case name (and the description), I would use `KrfCreationCanceledByCompaction`.
Test case name would then be:
`testCompactorRegionMapDeletedAfterCompactionForOnlyDrfOplogAndKrfCreationCanceledByCompactionIsPerformed()`
Also, I would change in the test name `IsDone` to `IsPerformed` to have the name aligned with the previous test case.